### PR TITLE
[maistra-2.5] OSSM-4056: Backport setting `istio.alpn_override: "false"` for SIMPLE and MUTUAL TLS

### PR DESCRIPTION
**Pull request by jewertow**
Don't merge it until we have maistra/proxy 2.5.

### CODE REVIEW

**Review by jshaughn — APPROVED** [review]
Just trusting @jewertow and unblocking this.  Up to you to wait for more approvals or merge.

**Review by yxun — APPROVED** [review]

**Pull request merged**
